code/Control/UserDefinedFormController.php

```diff
@@ -326,7 +326,7 @@ public function process($data, $form)
                 }
 
                 // check to see if they are a dynamic reply to. eg based on a email field a user selected
-                if ($recipient->SendEmailFromField()) {
+                if ($recipient->SendEmailFromField() && $recipient->SendEmailFromField()->exists()) {
```

Comment:
Could you please assign the result of `$recipient->SendEmailFromField()` to a variable, then using that instead of calling the getter twice?

Same with the other three instances :)